litestar/openapi/plugins.py

```diff
@@ -0,0 +1,584 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Sequence
+
+import msgspec
+import yaml
+
+from litestar.enums import MediaType, OpenAPIMediaType
+from litestar.handlers import get
+from litestar.serialization import encode_json, get_serializer
+
+if TYPE_CHECKING:
+    from litestar.connection import Request
+    from litestar.router import Router
+
+__all__ = (
+    "OpenAPIRenderPlugin",
+    "RapidocRenderPlugin",
+    "RedocRenderPlugin",
+    "StoplightRenderPlugin",
+    "SwaggerRenderPlugin",
+    "YamlRenderPlugin",
+)
+
+_favicon_url = "https://raw.githubusercontent.com/litestar-org/branding/main/assets/Branding%20-%20PNG%20-%20Transparent/Badge%20-%20Blue%20and%20Yellow.png"
+_default_favicon = f"<link rel='icon' type='image/png' href='{_favicon_url}'>"
+
+
+class OpenAPIRenderPlugin:
+    """Base class for OpenAPI UI render plugins."""
+
+    paths: list[str]
+
+    def __init__(
+        self,
+        *,
+        path: str | Sequence[str],
+        media_type: MediaType | OpenAPIMediaType = MediaType.HTML,
+        favicon: str = _default_favicon,
+        style: str = "body { margin: 0; padding: 0 }",
```

Comment:
It is inconsistent that we ask for a `<link>` tag for the favicon arg, but not a `<style>` tag for the style arg.

Previously, we only asked for a url for the favicon and constructed the `<link>` tag internally, however the `type=` attribute of the `<link>` tag for the favicon also needs to be specified depending on the the type of the image that is linked. So we modified it to accept the full `<link>`.

`<style>` tags similarly accept attributes so there is utility to accepting a complete `<style>` tag for the style arg beyond consistency.

Reply:
+1 for more consistency